docs/cli.rst

```diff
@@ -0,0 +1,287 @@
+information. The tool needs to be provided with keys and a :ref:`blocking-schema`; it will output a file containing
+json serialized candidate blocks.
+
+.. command-output:: anonlink block --help
+
+Example
+~~~~~~~
+
+Assume a csv (``fake-pii.csv``) contains rows like the following::
+
+
+    0,Libby Slemmer,1933/09/13,F
+    1,Garold Staten,1928/11/23,M
+    2,Yaritza Edman,1972/11/30,F
+
+It can be hashed using ``anonlink`` with::
+
+    $ anonlink block --schema blocking-schema.json fake-pii.csv horse candidate_blocks.json
+
+Describing
+----------
+
+Users can inspect the distribution of the number of bits set in ``CLKs`` by using the ``describe`` command.
+
+.. command-output:: anonlink describe --help
+
+
+Example
+~~~~~~~
+
+::
+
+    $ anonlink describe example_clks_a.json
+
+
+     339|                                   oo
+     321|                                  ooo
+     303|                                  ooo
+     285|                                  ooo o
+     268|                                  oooooo
+     250|                                oooooooo
+     232|                                oooooooo
+     214|                               ooooooooo
+     196|                             o ooooooooo o
+     179|                             o ooooooooooo
+     161|                             oooooooooooooo
+     143|                            ooooooooooooooo
+     125|                           oooooooooooooooo
+     107|                           oooooooooooooooooo
+      90|                         ooooooooooooooooooooo
+      72|                         oooooooooooooooooooooo
+      54|                        oooooooooooooooooooooooo
+      36|                      ooooooooooooooooooooooooooo
+      18|                   oooooooooooooooooooooooooooooooo
+       1| o  o  ooooooooooooooooooooooooooooooooooooooooooooooooooo oo
+         ------------------------------------------------------------
+         4 4 4 4 4 4 4 4 5 5 5 5 5 5 5 5 5 6 6 6 6 6 6 6 6 6 7 7 7 7
+         1 2 3 4 5 6 7 9 0 1 2 3 4 5 7 8 9 0 1 2 3 5 6 7 8 9 0 1 3 4
+         0 1 2 4 5 7 8 0 1 2 4 5 7 8 0 1 2 4 5 7 8 0 1 2 4 5 7 8 0 1
+           . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
+           4 8 3 7 1 6 0 4 9 3 7 2 6 0 5 9 3 8 2 6 1 5 9 4 8 2 7 1 5
+
+    -------------------------
+    |        Summary        |
+    -------------------------
+    |   observations: 5000  |
+    | min value: 410.000000 |
+    |   mean : 601.571600   |
+    | max value: 753.000000 |
+    -------------------------
+
+
+.. note::
+
+    It is an indication of problems in the hashing if the distribution is skewed towards no bits set or
+    all bits set. Consult the :doc:`tutorial/tutorial_cli` for further details.
+
+
+.. _schema_handling:
+
+Schema Handling
+---------------
+
+A schema file can be tested for validity against the schema specification with the ``validate-schema`` command.
```

Comment:
Does this work for a blocking schema as well as a linkage schema - probably worth indicating.